File: advent-of-code/2025/day-1-secret-entrance/normal/n-01-12.py
import sys

def solve(data):
    lines = data.strip().split('\n')
    pos = 50
    p1 = p2 = 0
    for line in lines:
        d = line[0]
        n = int(line[1:])
        for _ in range(n):
            if d == 'L':
                pos = (pos - 1) % 100
            else:
                pos = (pos + 1) % 100
            if pos == 0:
                p2 += 1
        if pos == 0:
            p1 += 1
    return p1, p2

if __name__ == "__main__": 
    p1, p2 = solve(sys.stdin. read())
    print(p1)
    print(p2)

File: advent-of-code/2025/day-1-secret-entrance/normal/n-01-12.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut pos:  i32 = 50;
    let (mut p1, mut p2) = (0, 0);
    for line in stdin.lock().lines().flatten() {
        let d = line.chars().next().unwrap();
        let n:  i32 = line[1..].parse().unwrap();
        for _ in 0.. n {
            pos = (pos + if d == 'L' { -1 } else { 1 }).rem_euclid(100);
            if pos == 0 { p2 += 1; }
        }
        if pos == 0 { p1 += 1; }
    }
    println!("{}\n{}", p1, p2);
}
